core/dbt/include/global_project/macros/materializations/models/table/table.sql

```diff
@@ -1,5 +1,6 @@
 {% materialization table, default %}
 
+  {% set original_query_tag = set_query_tag() %}
```

Comment:
We'll want this everywhere that `dbt-snowflake` currently has it implemented:
- table
- view
- incremental
- seed
- snapshot
- test
- MV / dynamic table

Then we can totally delete & remove the Snowflake-specific implementations of [test](https://github.com/dbt-labs/dbt-snowflake/blob/e1cde45c8b8ee7b4862e1a3c6417ad4f4e6df4b2/dbt/include/snowflake/macros/materializations/test.sql#L1-L8), [snapshot](https://github.com/dbt-labs/dbt-snowflake/blob/e1cde45c8b8ee7b4862e1a3c6417ad4f4e6df4b2/dbt/include/snowflake/macros/materializations/snapshot.sql), and [seed](https://github.com/dbt-labs/dbt-snowflake/blob/e1cde45c8b8ee7b4862e1a3c6417ad4f4e6df4b2/dbt/include/snowflake/macros/materializations/seed.sql) — which only exist to support setting/unsetting query tags.